velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -264,39 +282,51 @@ std::shared_ptr<const ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       switch (schemaElement.converted_type) {
         case thrift::ConvertedType::LIST:
         case thrift::ConvertedType::MAP: {
-          auto element = children.at(0)->getChildren();
           VELOX_CHECK_EQ(children.size(), 1);
-          return std::make_shared<const ParquetTypeWithId>(
+          return getElementTypeWithId(
+              children.at(0)->getChildren(),
               children[0]->type(),
-              std::move(element),
-              curSchemaIdx, // TODO: there are holes in the ids
+              curSchemaIdx,
+              name,
               maxSchemaElementIdx,
-              ParquetTypeWithId::kNonLeaf, // columnIdx,
-              std::move(name),
-              std::nullopt,
-              std::nullopt,
-              maxRepeat + 1,
+              maxRepeat,
               maxDefine);
         }
         case thrift::ConvertedType::MAP_KEY_VALUE: {
-          // child of MAP
-          VELOX_CHECK_EQ(
-              schemaElement.repetition_type,
-              thrift::FieldRepetitionType::REPEATED);
-          assert(children.size() == 2);
-          auto childrenCopy = children;
-          return std::make_shared<const ParquetTypeWithId>(
-              TypeFactory<TypeKind::MAP>::create(
-                  children[0]->type(), children[1]->type()),
-              std::move(childrenCopy),
-              curSchemaIdx, // TODO: there are holes in the ids
-              maxSchemaElementIdx,
-              ParquetTypeWithId::kNonLeaf, // columnIdx,
-              std::move(name),
-              std::nullopt,
-              std::nullopt,
-              maxRepeat,
-              maxDefine);
+          if (parentSchemaElement.converted_type !=
+              thrift::ConvertedType::MAP) {
+            // MAP_KEY_VALUE is incorrectly used in place of MAP
```

Comment:
seems this is assuming here that, if it is not thrift::ConvertedType::MAP, it must be a thrift::ConvertedType::MAP_KEY_VALUE? 

For more better robustness, to actually handle unexpected types, would it be better if we change it to explicit checking? e.g.

```

if (parentSchemaElement.converted_type == thrift::ConvertedType::MAP) {
    // do something...
} else if (parentSchemaElement.converted_type == thrift::ConvertedType::MAP_KEY_VALUE) {
    // do something
} else {
    // throw an unexpected type error
}
```

Reply:
I was trying to follow the [spec](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules-1) 

> For backward-compatibility, a group annotated with MAP_KEY_VALUE that is not contained by a MAP-annotated group should be handled as a MAP-annotated group.
As per spec the parent element cannot be MAP which is what we are checking here. Spec is not clear on what the actual converted_type should be.